test/authentication/middleware/index.spec.unit.ts

```diff
@@ -0,0 +1,75 @@
+import * as request from "supertest";
+import app from "../../../src/app";
+import {COOKIE_NAME} from "../../../src/properties";
+import * as keys from "../../../src/session/keys"
+import {loadSession} from "../../../src/services/redis.service";
+import {loadMockSession} from "../../mock.utils";
+import Session from "../../../src/session/session";
+
+jest.mock("../../../src/services/redis.service");
+
+const mockCacheService = (<unknown>loadSession as jest.Mock<typeof loadSession>);
+
+beforeEach(() => {
+  mockCacheService.mockRestore();
+  loadMockSession(mockCacheService);
+});
+
+describe("Authentication middleware", () => {
+
+  it("should not redirect to signin if loading start page", async () => {
+    const response = await request(app)
+      .get("/promise-to-file");
+    expect(response.status).toEqual(200);
+  });
+
+  it("should redirect to signin if /promise-to-file/* called and not signed in", async () => {
+    setNotSignedIn();
+    const response = await request(app)
+      .get("/promise-to-file/company-number")
+      .set("Referer", "/promise-to-file/company-number")
+      .expect("Location", "/signin?return_to=/promise-to-file");
+    expect(response.status).toEqual(302);
+  });
+
+  it("should redirect to signin if navigating from /promise-to-file page and not signed in", async () => {
+    setNotSignedIn();
+    const response = await request(app)
+      .get("/promise-to-file/company-number")
+      .set("Referer", "/promise-to-file")
+      .expect("Location", "/signin?return_to=/promise-to-file/company-number");
+    expect(response.status).toEqual(302);
+  });
+
+
+  it("should not redirect to signin if /promise-to-file/* called while signed in", async () => {
+    const response = await request(app)
+      .get("/promise-to-file/company-number")
+      .set("Referer", "/")
+      .set("Cookie", [`${COOKIE_NAME}=123`]);
+    expect(response.status).toEqual(200);
+  });
+
+  it("should redirect to start screen if /promise-to-file/* called from outside of the PTF journey and not signed in", async () => {
+    setNotSignedIn();
+    const response = await request(app)
+      .get("/promise-to-file/company-number")
+      .expect("Location", "/promise-to-file");
+    expect(response.status).toEqual(302);
+  });
+
+
```

Comment:
have you run tslint, i thought it moans about double blank lines.

Reply:
No. Will check.

Reply:
Right, tslint config looks slightly wrong because it's only picking up violations on the 'src' folder and not 'test'. We also want to ensure that test JS resources don't go in the 'dist' folder. @markpit to raise a new tech story to cover this work.